### ISSUE TITLE
Using KiTrack for forward STAR tracking

### ISSUE THREAD

**Issue by plexoos** [star-bnl/KiTrack#1]
I reviewed the proposed integration of KiTrack into the STAR framework as implemented in this repository https://github.com/klendathu2k/stroot-forward-tracker/tree/master/StRoot/StgUtil and here are a few comments:

- Hiding selected code (without the license, credits, and history) from the KiTrack package in StRoot/StgUtil does not look to me as a good idea. KiTrack seems well maintained and can be installed by itself with CMake. Perhaps, we should include it as an external package into our CMake-base STAR build.

- The differences between KiTrack and the proposed StRoot/StgUtil seem minor and I am not sure about their value. It is more valuable if we base our work on the official KiTrack release 

- I agree that for our purposes KiTrack does not need to depend on other packages in iLCSoft. It looks that this dependency can be easily broken and I will make the appropriate changes in this fork

@klendathu2k @jdbrice

**Comment by klendathu2k** [comment]
Agreed that it isn't ideal (just an easy starting point when compiling with "cons").

Daniel and I discussed differences between the official KiTrack and his modifications.
His statement was that they are necessary.  So for now, feel free to reorganize, but
keep the version which is in StgUtil.


On 10/30/19 2:41 PM, Dmitri Smirnov wrote:
> I reviewed the proposed integration of KiTrack into the STAR framework as implemented in this repository https://github.com/klendathu2k/stroot-forward-tracker/tree/master/StRoot/StgUtil and here are a few comments:
> 
> - Hiding selected code (without the license, credits, and history) from the KiTrack package in StRoot/StgUtil does not look to me as a good idea. KiTrack seems well maintained and can be installed by itself with CMake. Perhaps, we should include it as an external package into our CMake-base STAR build.
> 
> - The differences between KiTrack and the proposed StRoot/StgUtil seem minor and I am not sure about their value. It is more valuable if we base our work on the official KiTrack release
> 
> - I agree that for our purposes KiTrack does not need to depend on other packages in iLCSoft. It looks that this dependency can be easily broken and I will make the appropriate changes in this fork
> 
> @klendathu2k @jdbrice
> 

-- 
.--- .-- . -... -... .--.-. -... -. .-.. .-.-.- --. --- ...-

**Comment by plexoos** [comment]
I understand you are new to git and github... but forking someone's repository is not hard!
